flight/PiOS/STM32F30x/pios_servo.c

```diff
@@ -194,3 +194,75 @@ void PIOS_Servo_Set(uint8_t servo, uint16_t position)
 			break;
 	}
 }
+
+#if defined(PIOS_INCLUDE_ONESHOT)
+#define OneShotFrequency 12000000
+
+/**
+* Set servo position for OneShot
+* \param[in] Servo Servo number (0-num_channels)
+* \param[in] Position Servo position in 1/12 microseconds based on OneShotFrequency
+*/
+void PIOS_Servo_OneShot_Set(uint8_t servo, uint16_t position)
+{
+	/* Make sure servo exists */
+	if (!servo_cfg || servo >= servo_cfg->num_channels) {
+		return;
+	}
+
+	const struct pios_tim_channel * chan = &servo_cfg->channels[servo];
+
+	/* stop the timer */
+	TIM_Cmd(chan->timer, DISABLE);
```

Comment:
No, it is a hardware issue. There is one physical timer (that can support up to four outputs) and the way it behaves in OneShot isn't compatible with traditional PWM output or PWM/PPM input. This isn't a real problem, but is something we should make clear on the GUI to prevent people trying to put one channel on OneShot and having another behave strangely.